src/App.tsx

```diff
@@ -15,37 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+type GoodListProps = {
+  goods: string[];
+};
+
+enum SortField {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: { sortField: SortField | ''; reverse: boolean },
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortField.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+const GoodList: React.FC<GoodListProps> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li key={good} data-cy="Good">
+        {good}
+      </li>
+    ))}
+  </ul>
+);
```

Comment:
It's a good practice to define PropTypes or use TypeScript interfaces for the `GoodList` component to ensure that the expected props are passed and to catch potential errors during development.